File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/rubyNapakalaki/monster.rb
#encoding: utf-8

require_relative 'prize.rb'
require_relative 'bad_consequence.rb'

class Monster
  attr_accessor :name, :combat_level , :bad_consequence , :prize , :level_change_against_cultist_player

   def initialize(name, combat_level, bad_consequence, prize, ic)

    @name = name
    @combat_level= combat_level
    @bad_consequence = bad_consequence
    @prize = prize
    @level_change_against_cultist_player = ic

   end


   def self.copia(name, combat_level, bad_consequence, prize,ic)

      self.new(name, combat_level, bad_consequence, prize, ic)

   end

  def get_bad_consequence
    @bad_consequence
  end
  def get_combatlevel
    @combat_level
  end

  def get_levels_gained
    @prize.level
  end

  def get_treasures_gained
    @prize.treasures
  end

  def get_combat_level_against_cultist_player

        valor = get_combatlevel + @level_change_against_cultist_player
        valor
  end

  def to_s
    "\nNombre: #{@name}  Nivel de combate: #{@combat_level} \n#{@prize.to_s}  \n#{@bad_consequence.to_s}\n"
  end


 end

File: 2 SEGUNDO/1 Cuatrimestre/Programación y Diseño Orientado a Objetos - PDOO/Practicas/javaNapakalaki/NapakalakiGame/Monster.java
package NapakalakiGame;
import java.lang.*;

/**
 *
 * @author manjaro
 */

public class Monster {
    private String name;
    private int combatLevel;
    private BadConsequence badConse;
    private Prize prize;
    private int levelChangeAgainstCultistPlayer = 0 ;

    public Monster(String name, int level, BadConsequence bc, Prize prize){
        this.name = name;
        combatLevel = level;
        badConse = bc;
        this.prize = prize;
    }

    public Monster(String name, int level, BadConsequence bc, Prize prize, int IC){
        this.name = name;
        combatLevel = level;
        badConse = bc;
        this.prize = prize;
        this.levelChangeAgainstCultistPlayer = IC;

    }

    public String getName() {
        return name;
    }

    public int getCombatLevel() {
        return combatLevel;
    }

    public BadConsequence getBadConsequence() {
        return badConse;
    }

    public int getLevelsGained() {
        return prize.getLevel();
    }

    public int getTreasuresGained(){
        return prize.getTreasures();
    }

    public Prize getPrize(){
        return prize;
    }

    public int getCombatLevelAgainstCultistPlayer(){
        int res ;
        res = this.getCombatLevel() + this.levelChangeAgainstCultistPlayer;
        return res;
    }


    @Override
     public String toString(){

        String monstruo = "Nombre: " + name + " CombatLevel = " + Integer.toString(combatLevel)
                + "\nBad Consequence: " + badConse.toString() + "\nPrize: " + prize.toString() + "\n";
        return monstruo;
      }

}
